dsc/src/resource_command.rs

```diff
@@ -55,12 +55,24 @@ pub fn get(dsc: &DscManager, resource_type: &str, input: &Option<String>, stdin:
 }
 
 pub fn get_all(dsc: &DscManager, resource_type: &str, _input: &Option<String>, _stdin: &Option<String>, format: &Option<OutputFormat>) {
-    let Some(resource) = get_resource(dsc, resource_type) else {
+    let mut input = String::new() ;
+    let Some(mut resource) = get_resource(dsc, resource_type) else {
         error!("{}", DscError::ResourceNotFound(resource_type.to_string()).to_string());
         return
     };
+
     debug!("resource.type_name - {} implemented_as - {:?}", resource.type_name, resource.implemented_as);
-    let export_result = match resource.export() {
+    if let Some(requires) = &resource.requires {
+        input = add_type_name_to_json(input, resource.type_name.clone());
+        if let Some(pr) = get_resource(dsc, requires) {
+            resource = pr;
+        } else {
+            error!("Provider {} not found", requires);
+            return;
+        };
+    }
+
+    let export_result = match resource.export(input.as_str()) {
```

Comment:
Use ref syntax instead of calling the function for consistency

```suggestion
    let export_result = match resource.export(&input) {
```